src/app/pages/sharing/components/shares-dashboard/shares-dashboard.component.ts

```diff
@@ -605,4 +621,24 @@ export class SharesDashboardComponent implements AfterViewInit {
         return 'fn-theme-orange';
     }
   }
+
+  hideColumnIfEmpty(column: string, type: ShareType, data: SmbShare[] | WebDavShare[] | NfsShare[]): void {
+    switch (type) {
+      case ShareType.SMB:
+        this.smbTableConf.columns.find((item) => item.prop === column).hidden = !(data as SmbShare[]).some((item) =>
```

Comment:
We should break this checks down for better readability and understanding.

Reply:
That's a good point. I moved this feature to the `TableComponent` itself in the next PR. Instead of implementing for each dashboard, we'll need to pass `hiddenIfEmpty: true` on column config, and the table component will handle it.